lib/mpl_toolkits/mplot3d/axis3d.py

```diff
@@ -516,21 +516,44 @@ def v_interval(self, minmax):
 # Use classes to look at different data limits
 
 
-class XAxis(Axis):
+class XAxis(Axis, maxis.XAxis):
     get_view_interval, set_view_interval = maxis._make_getset_interval(
         "view", "xy_viewLim", "intervalx")
     get_data_interval, set_data_interval = maxis._make_getset_interval(
         "data", "xy_dataLim", "intervalx")
 
 
-class YAxis(Axis):
+class YAxis(Axis, maxis.YAxis):
     get_view_interval, set_view_interval = maxis._make_getset_interval(
         "view", "xy_viewLim", "intervaly")
     get_data_interval, set_data_interval = maxis._make_getset_interval(
         "data", "xy_dataLim", "intervaly")
 
-
-class ZAxis(Axis):
+    def get_tick_space(self):
+        ends = mtransforms.Bbox.from_bounds(0, 0, 1, 1)
+        ends = ends.transformed(self.axes.transAxes -
+                                self.figure.dpi_scale_trans)
+        length = ends.height * 72
+        # The spacing must be the same as in the XAxis method (3).
+        size = self._get_tick_label_size('y') * 3
```

Comment:
AFAICS, using 'y'  is the only difference to before. In that case, shouldn't we fix the original `Axis.get_tickspace` to use `self._get_tick_label_size(self.axis_name)` (and define `axis_name` on the 3d Axis classes)?

Reply:
Well, the only difference to before is in fact not the use of 'y' but the multiplication by 3. We can see in the axis.py file that the method `get_tick_space` in the XAxis class has the line `size = self._get_tick_label_size('x') * 3`, whereas in the YAxis class the same method has the line `size = self._get_tick_label_size('y') * 2`

The reason for this change in coefficient is the aspect of a 2D plot : it requires less space to write labels like this :
```
label    (Y labels)
label
label 
```
Than like this :
`label label label  (X labels)`

However, in 3D plots the Y axis is like another X axis : hence the overriding of the method, even though it's just for one digit.

Reply:
Thanks for the clarification. It seems we should turn the factor into a class variable then.

Reply:
You're probably right, it would be more convenient. Oh and I also had a question : would you know why the`codecov/progect/tests` omitted the `src` file in this PR ? I'm new to github and matplotlib, I don't really understand how all of it works.

Reply:
I have to admit, that I don't understand Codecov either. It sometimes appears to do weird things.